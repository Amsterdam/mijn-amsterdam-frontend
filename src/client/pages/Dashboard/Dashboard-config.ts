export const themaId = 'HOME';
export const themaTitle = 'Home';

export const DASHBOARD_PAGE_DOCUMENT_TITLE = 'Home | Mijn Amsterdam';
