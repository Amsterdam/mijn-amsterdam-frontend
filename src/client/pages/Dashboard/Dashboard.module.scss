@import '../../styles/_global.scss';

.Dashboard {
  // Dashboard has a different design for the heading, to match the design easily and responsively
  // a pragmatic solution with help of :before is written below.
  @include select(PageHeading_PageHeading) {
    @include centered-max;
    background: transparent;
    margin-bottom: 0;

    &:before {
      content: '';
      position: absolute;
      background: #fff;
      width: calc(50% - 2rem);
      left: 0;
      top: 0;
      bottom: 0;
    }

    @include mq-phone() {
      margin: 0 1rem;
    }

    @include select(PageHeading_HeadingInner) {
      margin: 0 4rem;
      z-index: 1;

      @include mq-tablet() {
        margin: 0 2rem;
      }

      @include mq-phone() {
        margin: 0;
      }
    }
  }

  .MyNotificationsHeadingLink {
    text-decoration: none;
    &:hover {
      text-decoration: underline;
      color: $color-primary-red;
    }
  }

  .TopContentContainer {
    @include centered-max;

    @include media-q('>phone') {
      display: flex;
      align-items: flex-start;
    }
  }

  @include select(MyNotifications_MyNotifications) {
    overflow: hidden;
    border-top: 1px solid $color-neutral-grey2;
    margin: 0 1rem;

    @include media-q('>phone') {
      flex-grow: 1;
      margin: 0 2rem;
      margin-bottom: 8rem;
    }

    @include media-q('>tablet') {
      margin-bottom: 8rem;
    }
  }

  @include select(MyChaptersPanel_MyChaptersPanel) {
    // Adjust some margins to match the design on mobile
    @include media-q('>phone') {
      flex: 0 0 calc(50% + 2rem);
    }
  }

  @include select(MyArea_MyAreaMapContainer) {
    height: 48rem;
    margin-bottom: 10rem;
    width: 100%;
  }

  @include select(DirectLinks_DirectLinks) {
    @include centered-block;
  }

  @include select(MyTips_MyTips) {
    @include centered-block;
    margin-bottom: 10rem;
  }
}
