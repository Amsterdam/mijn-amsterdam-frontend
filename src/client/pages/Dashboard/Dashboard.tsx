import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { AppRoutes } from '../../../universal/config';
import { isLoading } from '../../../universal/helpers';
import { AppContext } from '../../AppState';
import {
  DirectLinks,
  MyAreaDashboard,
  MyCases,
  MyChaptersPanel,
  MyNotifications,
  MyTips,
  Page,
  PageHeading,
} from '../../components';
import { getMyChapters } from '../../helpers/chapters';
import { usePhoneScreen } from '../../hooks/media.hook';
import styles from './Dashboard.module.scss';

const MAX_NOTIFICATIONS_VISIBLE = 3;
const MAX_TIPS_VISIBLE = 3;

export default () => {
  const appState = useContext(AppContext);
  const { TIPS, NOTIFICATIONS, CASES, BUURT, HOME } = appState;

  const tipItems = TIPS.content?.slice(0, MAX_TIPS_VISIBLE) || [];
  const notificationItems =
    NOTIFICATIONS.content?.slice(0, MAX_NOTIFICATIONS_VISIBLE) || [];
  const isPhoneScreen = usePhoneScreen();
  const NOTIFICATIONSTotal = NOTIFICATIONS.content?.length || 0;

  const {
    items: myChapterItems,
    isLoading: isMyChaptersLoading,
  } = getMyChapters(appState);

  return (
    <>
      <Page
        className={styles.Dashboard}
        data-tutorial-item="Hier ziet u nieuwe berichten van onze afdelingen die uw aandacht vragen;right-top"
      >
        <PageHeading>
          <Link
            className={styles.MyNotificationsHeadingLink}
            to={AppRoutes.NOTIFICATIONS}
          >
            Actueel
          </Link>
        </PageHeading>
        <div className={styles.TopContentContainer}>
          <MyNotifications
            items={notificationItems}
            showMoreLink={NOTIFICATIONSTotal > MAX_NOTIFICATIONS_VISIBLE}
            isLoading={isLoading(NOTIFICATIONS)}
            trackCategory="Dashboard / Actueel"
          />
          <MyChaptersPanel
            isLoading={isMyChaptersLoading}
            items={myChapterItems}
            title="Mijn thema's"
            data-tutorial-item="Dit zijn de onderwerpen waarover u iets heeft bij de gemeente;right-top"
            trackCategory="Dashboard / Mijn Thema's"
          />
        </div>

        <MyCases
          isLoading={isLoading(CASES)}
          title="Mijn lopende aanvragen"
          data-tutorial-item="Dit is een overzicht van uw lopende aanvragen of wijzigingen;right-top"
          items={CASES.content!}
        />

        {!isPhoneScreen && (
          <MyAreaDashboard
            url={BUURT.content?.embed.simple}
            center={HOME.content?.latlng}
            data-tutorial-item="Hier ziet u informatie van de gemeente, bijvoorbeeld over afval, parkeren en bekendmakingen;left-top"
          />
        )}

        {!isPhoneScreen && (
          <MyTips
            data-tutorial-item="Hier geven wij u handige tips, bijvoorbeeld over de regelingen en voorzieningen van de gemeente;right-bottom"
            isLoading={isLoading(TIPS)}
            items={tipItems}
          />
        )}
        <DirectLinks data-tutorial-item="Hier kunt u meer algemene informatie vinden over Mijn Amsterdam, bijvoorbeeld waar u terecht kunt met uw vragen;right-bottom" />
      </Page>
    </>
  );
};
