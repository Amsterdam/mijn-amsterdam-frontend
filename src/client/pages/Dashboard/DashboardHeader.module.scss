.DashboardHeaderInner {
  position: relative;
  padding-top: 31.25%;
  overflow: hidden;
  background-color: var(--ams-color-separator);
}
@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
.DashboardHeaderImg {
  top: 0;
  height: 100%;
  display: block;
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
}
.fadeIn {
  animation: fadeIn 0.3s ease-out;
}
.DashboardHeaderSearch {
  position: absolute;
  bottom: var(--ams-space-xl);
  left: 50%;
  transform: translateX(-50%);
  width: calc(100% - var(--ams-space-l) * 2);
  @media screen and (min-width: $ams-breakpoint-medium) {
    width: 63%;
  }
}
