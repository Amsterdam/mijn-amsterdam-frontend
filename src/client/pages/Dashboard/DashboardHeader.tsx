import { useEffect, useRef, useState } from 'react';

import classNames from 'classnames';

import styles from './DashboardHeader.module.scss';
import { Search } from '../../components/Search/Search';
import { usePhoneScreen } from '../../hooks/media.hook';
import { useAppStateGetter } from '../../hooks/useAppState';
import { useProfileTypeValue } from '../../hooks/useProfileType';

const STADSDELEN = [
  'buiten-amsterdam',
  'centrum',
  'nieuw-west',
  'noord',
  'oost',
  'weesp',
  'west',
  'westpoort',
  'zuid',
  'zuidoost',
];

function useStadsdeelFoto() {
  const { MY_LOCATION } = useAppStateGetter();
  const isPhoneScreen = usePhoneScreen();
  const [primaryLocation = null] = MY_LOCATION.content ?? [];
  const profileType = useProfileTypeValue();

  let stadsdeel =
    primaryLocation?.bagAddress?.gebiedenStadsdeelNaam?.toLowerCase();

  if (!stadsdeel || !STADSDELEN.includes(stadsdeel)) {
    stadsdeel = `buiten-amsterdam-${profileType}`;
  }

  const size = isPhoneScreen ? 'small' : 'large';

  return {
    srcDefault: `/img/stadsdeel-foto/${size}/buiten-amsterdam-${profileType}.jpg`,
    src: stadsdeel ? `/img/stadsdeel-foto/${size}/${stadsdeel}.jpg` : null,
  };
}

export function DashboardHeader() {
  const { src, srcDefault } = useStadsdeelFoto();
  const [isLoading, setIsLoading] = useState(true);
  const imgRef = useRef<HTMLImageElement | null>(null);

  useEffect(() => {
    if (!src) {
      return;
    }
    if (!imgRef.current) {
      const img = new Image();
      imgRef.current = img;
      img.src = src;
      img.onload = () => setIsLoading(false);
      img.onerror = () => setIsLoading(false);
    } else if (imgRef.current.src !== src) {
      imgRef.current.src = src;
      setIsLoading(true);
    }
  }, [src]);

  return (
    <header className={styles.DashboardHeader}>
      <div className={styles.DashboardHeaderInner}>
        {!!imgRef.current?.src && !isLoading && (
          <img
            className={classNames(styles.DashboardHeaderImg, styles.fadeIn)}
            src={imgRef.current.src}
            alt=""
          />
        )}
        <img
          className={classNames(styles.DashboardHeaderImg, styles.default)}
          src={srcDefault}
          alt=""
        />
        <div className={styles.DashboardHeaderSearch}>
          <Search />
        </div>
      </div>
    </header>
  );
}
