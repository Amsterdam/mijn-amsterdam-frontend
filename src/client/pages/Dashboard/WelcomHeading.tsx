import styles from './WelcomeHeading.module.scss';
import { ProfileName } from '../../components/MainHeader/ProfileName';
import { PageHeadingV2 } from '../../components/PageHeading/PageHeadingV2';

const HELLO = 'Welkom,';
const NOON = 12;
const EVENING = 18;
const NIGHT = 5;

export function WelcomeHeading() {
  const hours = new Date().getHours();
  const isMorning = hours < NOON && hours >= NIGHT;
  const isEvening = hours >= EVENING;
  const isNight = hours < NIGHT;
  const isAfternoon = hours >= NOON && hours < EVENING;

  let hello = HELLO;

  switch (true) {
    case isMorning:
      hello = 'Goedemorgen,';
      break;
    case isAfternoon:
      hello = 'Goedemiddag,';
      break;
    case isEvening:
      hello = 'Goedenavond,';
      break;
    case isNight:
      hello = 'Goedenacht,';
      break;
  }

  return (
    <PageHeadingV2 showBacklink={false}>
      {hello}{' '}
      <span className={styles.ProfileNameWrap}>
        <ProfileName />
      </span>
    </PageHeadingV2>
  );
}
