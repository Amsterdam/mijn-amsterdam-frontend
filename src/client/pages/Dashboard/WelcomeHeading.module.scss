.WelcomeHeadingWrap {
  background-color: white;
  position: relative;

  // Hack to add white background to the left of the screen.
  &:before {
    content: '';
    position: absolute;
    background-color: white;
    top: 0;
    left: -100%;
    height: 100%;
    width: 100%;
    display: block;
  }

  @media screen and (min-width: $ams-breakpoint-medium) {
    margin-top: -166px;
  }
}

.WelcomeHeadingSizer {
  position: relative;
  @media screen and (min-width: $ams-breakpoint-medium) {
    height: 166px;
  }
}

@media screen and (min-width: $ams-breakpoint-medium) {
  .WelcomeHeading {
    // Align bottom of text with rest of page.
    position: relative;
    top: 100%;
    transform: translateY(calc(-100% + 18px));
    line-height: 0;
    margin: 0;
  }
}

.ProfileNameHello {
  display: block;
  line-height: 1.5;
  @media screen and (min-width: $ams-breakpoint-medium) {
    font-size: 58px;
  }
}

.ProfileNameWrap {
  font-size: 40px;
  line-height: 1;
  @media screen and (min-width: $ams-breakpoint-medium) {
    white-space: nowrap;
    line-height: 1;
    font-size: 80px;
    overflow: hidden;
    display: block;
    text-overflow: ellipsis;
  }
}

:global(.ams-link--standalone) {
  &.BackLink {
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    gap: 10px;
  }
}
