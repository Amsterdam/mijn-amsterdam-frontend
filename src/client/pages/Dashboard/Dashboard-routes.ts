import { Dashboard } from './Dashboard.tsx';
import { themaTitle, themaId } from './Dashboard-config.ts';

export const DashboardRoute = {
  route: '/',
  Component: Dashboard,
  props: { index: true },
  public: false,
  private: true,
};

export const DashboardRoutes = [DashboardRoute];

export const dashboardMenuItem = {
  title: themaTitle,
  id: themaId,
  to: DashboardRoute.route,
};
