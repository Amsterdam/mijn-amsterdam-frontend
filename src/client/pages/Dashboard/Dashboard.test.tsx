import { render } from '@testing-library/react';
import { generatePath } from 'react-router-dom';
import { MutableSnapshot, RecoilState } from 'recoil';
import { describe, expect, it } from 'vitest';

import { AppRoutes } from '../../../universal/config/routes';
import { Themas } from '../../../universal/config/thema';
import { AppState } from '../../../universal/types/App.types';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import Dashboard from './Dashboard';
import { remoteApiHost } from '../../../testing/setup';

const testState = {
  BRP: {
    status: 'OK',
    content: {
      mokum: true,
    },
  },
  MY_LOCATION: { status: 'OK', content: [{ latlng: { lat: 5, lng: 40 } }] },
  NOTIFICATIONS: {
    status: 'OK',
    content: [
      {
        id: 'Not1',
        title: 'Notification',
        description: 'Notificatie1',
        datePublished: '2020-07-24',
        thema: Themas.ROOT,
        link: {
          to: '/item-1',
          title: 'Linkje!',
        },
      },
      {
        id: 'Not2',
        title: 'Notification',
        description: 'Notificatie2',
        datePublished: '2020-07-24',
        thema: Themas.ROOT,
        link: {
          to: '/item-2',
          title: 'Linkje!',
        },
      },
      {
        id: 'Not3',
        title: 'Notification',
        description: 'Notificatie3',
        datePublished: '2020-07-24',
        thema: Themas.ROOT,
        isAlert: true,
        link: {
          to: '/item-3',
          title: 'Linkje!',
        },
      },
    ],
  },

  // Some themas
  BELASTINGEN: {
    content: {
      isKnown: true,
    },
  },
  MILIEUZONE: {
    content: {
      isKnown: true,
    },
  },
  VERGUNNINGEN: {
    isActive: true,
  },
  VERGUNNINGENv2: {
    isActive: true,
  },
  PARKEREN: {
    content: {
      isKnown: true,
      url: `${remoteApiHost}/sso/portaal/parkeren`,
    },
  },
  WPI_TOZO: {
    content: [{}],
  },
} as unknown as AppState;

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom as RecoilState<Partial<AppState>>, testState);
}

describe('<Dashboard />', () => {
  const routeEntry = generatePath(AppRoutes.ROOT);
  const routePath = AppRoutes.ROOT;

  function Component() {
    return (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={Dashboard}
        initializeState={initializeState}
      />
    );
  }

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
