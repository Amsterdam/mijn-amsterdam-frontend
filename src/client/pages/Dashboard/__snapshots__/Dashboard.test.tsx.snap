// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="Page Dashboard"
  >
    <header
      class="PageHeading"
    >
      <div
        class="HeadingInner"
      >
        <h2
          class="Heading Heading__large"
        >
          <a
            class="MyNotificationsHeadingLink"
            href="/overzicht-updates"
          >
            Actueel
          </a>
        </h2>
      </div>
    </header>
    <div
      class="TopContentContainer"
    >
      <div
        class="MyNotifications isLoading"
      >
        <ul>
          <li
            class="MyNotificationItem"
          >
            <aside
              class="MetaInfo"
            >
              <button
                aria-expanded="false"
                class="TitleToggle"
              >
                <h4
                  class="Heading Heading__small Title NotificationHeader"
                >
                  Notification
                </h4>
                <svg
                  aria-label="Home"
                  class="ChapterIcon Icon"
                  fill="#EC0000"
                >
                  burgerzaken.svg
                </svg>
                <div
                  class="MetaInfoSecondary"
                >
                  <em
                    class="ChapterIndication"
                  >
                    Home
                  </em>
                  <time
                    class="Datum"
                    datetime="2020-07-24"
                  >
                    24 juli 2020
                  </time>
                </div>
                <svg
                  aria-hidden="true"
                  class="CaretIcon"
                >
                  Chevron-Right.svg
                </svg>
              </button>
            </aside>
            <div
              aria-hidden="true"
              class="Body"
              style="height: 0px;"
            >
              <div>
                <div
                  class="Description"
                >
                  Notificatie1
                </div>
                <p
                  class="Action"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/item-1"
                    title="Meer informatie over de melding: Notification"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Linkje!
                  </a>
                </p>
              </div>
            </div>
          </li>
          <li
            class="MyNotificationItem"
          >
            <aside
              class="MetaInfo"
            >
              <button
                aria-expanded="false"
                class="TitleToggle"
              >
                <h4
                  class="Heading Heading__small Title NotificationHeader"
                >
                  Notification
                </h4>
                <svg
                  aria-label="Home"
                  class="ChapterIcon Icon"
                  fill="#EC0000"
                >
                  burgerzaken.svg
                </svg>
                <div
                  class="MetaInfoSecondary"
                >
                  <em
                    class="ChapterIndication"
                  >
                    Home
                  </em>
                  <time
                    class="Datum"
                    datetime="2020-07-24"
                  >
                    24 juli 2020
                  </time>
                </div>
                <svg
                  aria-hidden="true"
                  class="CaretIcon"
                >
                  Chevron-Right.svg
                </svg>
              </button>
            </aside>
            <div
              aria-hidden="true"
              class="Body"
              style="height: 0px;"
            >
              <div>
                <div
                  class="Description"
                >
                  Notificatie2
                </div>
                <p
                  class="Action"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/item-2"
                    title="Meer informatie over de melding: Notification"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Linkje!
                  </a>
                </p>
              </div>
            </div>
          </li>
          <li
            class="MyNotificationItem"
          >
            <aside
              class="MetaInfo"
            >
              <button
                aria-expanded="false"
                class="TitleToggle"
              >
                <h4
                  class="Heading Heading__small Title NotificationHeader"
                >
                  Notification
                </h4>
                <svg
                  aria-label="ALERT"
                  class="ChapterIcon Icon"
                  fill="#EC0000"
                >
                  Alert.svg
                </svg>
                <div
                  class="MetaInfoSecondary"
                >
                  <em
                    class="ChapterIndication"
                  >
                    Home
                  </em>
                  <time
                    class="Datum"
                    datetime="2020-07-24"
                  >
                    24 juli 2020
                  </time>
                </div>
                <svg
                  aria-hidden="true"
                  class="CaretIcon"
                >
                  Chevron-Right.svg
                </svg>
              </button>
            </aside>
            <div
              aria-hidden="true"
              class="Body"
              style="height: 0px;"
            >
              <div>
                <div
                  class="Description"
                >
                  Notificatie3
                </div>
                <p
                  class="Action"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/item-3"
                    title="Meer informatie over de melding: Notification"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Linkje!
                  </a>
                </p>
              </div>
            </div>
          </li>
        </ul>
        <p
          class="FooterLink"
        >
          <a
            class="Button Button__plain Button__lean Linkd"
            href="/overzicht-updates"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              Chevron-Right.svg
            </svg>
            Alle updates
          </a>
        </p>
      </div>
      <div
        class="Panel MyChaptersPanel"
      >
        <div
          class="Header"
        >
          <h3
            class="Heading Heading__large Title"
          >
            Mijn thema's
          </h3>
          <a
            aria-label="Dit ziet u in Mijn Amsterdam"
            class="Button Button__plain Button__lean Linkd GeneralInfoLink"
            href="/uitleg"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              Info.svg
            </svg>
          </a>
        </div>
        <div
          class="Links"
        >
          <a
            class="MainNavSubmenuLink"
            data-chapter-id="BELASTINGEN"
            href="https://belastingbalie.amsterdam.nl/digid.saml.php?start"
            rel="external"
          >
            <span
              class="SubmenuItemIcon"
            >
              <svg
                aria-hidden="true"
                fill="#767676"
              >
                belastingen.svg
              </svg>
            </span>
            <span
              class="SubmenuItemTitle"
            >
              Belastingen
            </span>
          </a>
          <a
            class="MainNavSubmenuLink"
            data-chapter-id="INKOMEN"
            href="/inkomen"
          >
            <span
              class="SubmenuItemIcon"
            >
              <svg
                aria-hidden="true"
                fill="#767676"
              >
                inkomen.svg
              </svg>
            </span>
            <span
              class="SubmenuItemTitle"
            >
              Inkomen
            </span>
          </a>
          <a
            class="MainNavSubmenuLink"
            data-chapter-id="MILIEUZONE"
            href=""
            rel="external"
          >
            <span
              class="SubmenuItemIcon"
            >
              <svg
                aria-hidden="true"
                fill="#767676"
              >
                milieuzone.svg
              </svg>
            </span>
            <span
              class="SubmenuItemTitle"
            >
              Milieuzone
            </span>
          </a>
          <a
            class="MainNavSubmenuLink"
            data-chapter-id="PARKEREN"
            href="/parkeren-en-verkeer"
          >
            <span
              class="SubmenuItemIcon"
            >
              <svg
                aria-hidden="true"
                fill="#767676"
              >
                Car.svg
              </svg>
            </span>
            <span
              class="SubmenuItemTitle"
            >
              Parkeren en verkeer
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="DashboardMapContainer"
    >
      <div
        class="MyAreaLoader"
      >
        Loading buurt bundle...
      </div>
      <a
        class="NavLink"
        href="/buurt"
      >
        <span
          class="NavLinkContentWrap"
        >
          <h3
            class="Heading Heading__large"
          >
            Mijn buurt
          </h3>
          <p>
            Klik voor een overzicht van gemeentelijke informatie rond uw eigen woning.
          </p>
        </span>
      </a>
    </div>
    <div
      class="MyTips"
    >
      <div
        class="HeaderBar"
      >
        <h3
          class="Heading Heading__large"
        >
          Mijn tips
        </h3>
        <button
          aria-expanded="false"
          class="Button Button__plain Button__lean OptIn"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            Chevron-Right.svg
          </svg>
          Maak tips persoonlijk
        </button>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="/overzicht-tips"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            Chevron-Right.svg
          </svg>
          Al mijn tips
        </a>
      </div>
      <ul
        class="TipsList"
      >
        <li
          class="TipItem is-personalized"
        >
          <article>
            <div
              class="ImageContainer"
            >
              <img
                alt=""
                class="Img"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAMLCwgAAACH5BAAAAAAALAAAAAABAAEAAAICRAEAOw=="
              />
              <div>
                <button
                  aria-expanded="false"
                  aria-label="Reden waarom u deze tip ziet"
                  class="Button Button__plain Button__lean IconButton TipFlipButton"
                  id="tip-flip-Tip1"
                >
                  <svg
                    aria-hidden="true"
                    class="Icon"
                    fill="#ffffff"
                    height="32"
                    width="32"
                  >
                    Info.svg
                  </svg>
                </button>
                <p
                  aria-labelledby="tip-flip-Tip1"
                  class="TipFlip"
                  hidden=""
                >
                  <span>
                    U ziet deze tip omdat!
                  </span>
                </p>
              </div>
            </div>
            <h4
              class="Heading Heading__medium"
            >
              Tip!!
            </h4>
            <p>
              Tip over dingen
            </p>
            <a
              class="Button Button__plain Button__lean Linkd"
              href="/lopende-zaak-1"
              title="Meer informatie over de tip: Tip!!"
            >
              <svg
                aria-hidden="true"
                class="Icon Icon__left"
                fill="#000"
                height="14px"
                width="14px"
              >
                Chevron-Right.svg
              </svg>
              Linkje!
            </a>
          </article>
        </li>
        <li
          class="TipItem"
        />
      </ul>
    </div>
    <div
      class="DirectLinks"
      id="DirectLinks"
    >
      <h3
        class="Heading Heading__large"
        id="DirectLinksHeader"
      >
        Bezoek ook
      </h3>
      <ul
        class="LinkList"
      >
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="/buurt"
            id="MyAreaHeader"
          >
            Mijn buurt
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="/overzicht-tips"
            id="MyTipsHeader"
          >
            Mijn tips
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="https://mijnsubsidies.amsterdam.nl/loket/"
            rel="external noopener noreferrer"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              External-Link.svg
            </svg>
            Mijn Subsidies
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="https://www.amsterdam.nl/sbl/"
            rel="external noopener noreferrer"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              External-Link.svg
            </svg>
            Stadsbank van Lening
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="https://www.amsterdam.nl/veelgevraagd/?productid={68422ECA-8C56-43EC-A9AA-B3DF190B5077}"
            rel="external noopener noreferrer"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              External-Link.svg
            </svg>
            Veelgestelde vragen
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="https://mijn.overheid.nl/berichtenbox/inbox/"
            rel="external noopener noreferrer"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              External-Link.svg
            </svg>
            MijnOverheid Berichtenbox
          </a>
        </li>
        <li>
          <a
            class="Button Button__plain Button__lean Linkd"
            href="https://www.amsterdam.nl"
            rel="external noopener noreferrer"
          >
            <svg
              aria-hidden="true"
              class="Icon Icon__left"
              fill="#000"
              height="14px"
              width="14px"
            >
              External-Link.svg
            </svg>
            Amsterdam.nl
          </a>
        </li>
      </ul>
    </div>
  </main>
</DocumentFragment>
`;
